pkg/proc/gdbserial/gdbserver_conn.go

```diff
@@ -761,10 +773,45 @@ func (conn *gdbConn) parseStopPacket(resp []byte, threadID string, tu *threadUpd
 		var metype int
 		medata := make([]uint64, 0, 10)
 
+		parseMachException := func(sp *stopPacket, metype int, medata []uint64) {
+			// Debugserver does not report watchpoint stops in the standard way preferring
+			// instead the semi-undocumented metype/medata keys.
+			// These values also have different meanings depending on the CPU architecture.
+			switch conn.goarch {
+			case "amd64":
+				if metype == _EXC_BREAKPOINT && len(medata) >= 2 && medata[0] == _EXC_I386_SGL {
+					sp.watchAddr = medata[1] // this should be zero if this is really a single step stop and non-zero for watchpoints
+				}
+			case "arm64":
+				if metype == _EXC_BREAKPOINT && len(medata) >= 2 && (medata[0] == _EXC_ARM_DA_DEBUG || medata[0] == _EXC_ARM_BREAKPOINT) {
+					// The arm64 specification allows for up to 16 debug registers.
+					// The registers are zero indexed, thus a value less than 16 will
+					// be a hardware breakpoint register index.
+					// See: https://developer.arm.com/documentation/102120/0101/Debug-exceptions
+					// TODO(deparker): we can ask debugserver for the number of hardware breakpoints
+					// directly.
+					if medata[1] < 16 {
+						sp.watchReg = int(medata[1])
+					} else {
+						sp.watchAddr = medata[1]
+					}
+				}
+			}
+		}
+
 		csp := colonSemicolonParser{buf: resp[3:]}
 		for csp.next() {
 			key, value := csp.key, csp.value
 
+			if reg, err := strconv.ParseUint(string(key), 16, 64); err == nil {
```

Comment:
Last time I checked this only had a few registers but we can't make use of only a few registers, we only know how to have all of them or none of them.

Reply:
I think the easiest thing to do would be to use the info from the stop packet and then load every register that isn't there by using `readRegister`.